File: zmy/cpp/1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */
class Solution {
public:
    std::vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> mp;
        for (auto i = std::begin(nums); i!= std::end(nums); i++) {
            if (mp.find(target - (*i)) != std::end(mp)) {
                return {mp[target - (*i)], std::distance(std::begin(nums), i)};
            } else {
                mp[*i] = std::distance(std::begin(nums), i);
            }
        }
        return {};
    }
};

File: zmy/golang/1.two-sum.go
/*
 * @lc app=leetcode id=1 lang=golang
 *
 * [1] Two Sum
 */
func twoSum(nums []int, target int) []int {
	mp := make(map[int]int)
	for index, value := range nums {
		if i, ok := mp[target - value]; ok {
			return []int{i, index}
		} else {
			mp[value] = index
		}
	}
	return []int{}
}

